don't show colormap for rgb images

ekdnam opened the pull request:
# Description
<!-- What does this pull request (PR) do? Why is it necessary? -->
<!-- Tell us about your new feature, improvement, or fix! -->
Solves issue: https://github.com/napari/napari/issues/1542

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
<!-- What resources, documentation, and guides were used in the creation of this PR? -->
<!-- If this is a bug-fix or otherwise resolves an issue, reference it here with "closes #(issue)" -->

# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

ekdnam committed:
use layer.rgb to not show colormap

ekdnam committed:
autopep8 applied to pass flake8 linting

ekdnam committed:
apply black formatting

ekdnam committed:
resolve colorbar widget error

now, the colorbar widget does not appear at the top left corner when the image has a rgb layer

sofroniewn commented on napari/_qt/layer_controls/qt_image_controls.py:
```diff
@@ -98,8 +98,12 @@ def __init__(self, layer):
+        # The colormap widget won't be created when image has a rgb layer
+        if layer.rgb is False:
+            self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
+            self.grid_layout.addLayout(colormap_layout, 3, 1)
```
I'd undo these changes, I think if we are going with the hide approach it is fine to add them and then hide them.

sofroniewn commented on napari/_qt/layer_controls/qt_image_controls.py:
```diff
@@ -98,8 +98,12 @@ def __init__(self, layer):
+
+        # The colormap widget won't be created when image has a rgb layer
+        if layer.rgb is False:
+            self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
```
let's actually store the reference to the colormap text label explicitly by making `self.colormapLabel = QLabel('colormap:')`. Can you do that here https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/qt_image_controls_base.py#L56

And then here can call `self.grid_layout.addWidget(self.colormapLabel, 3, 0)`

You can also then change this line accordingly too https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/qt_surface_controls.py#L39

Having this reference to the QLabel will make hiding the label much cleaner

sofroniewn commented on napari/_qt/layer_controls/qt_image_controls.py:
```diff
@@ -114,6 +118,11 @@ def __init__(self, layer):
 
+        # Similarly, the colormap combo box will be hidden if the layer is rgb
+        if layer.rgb:
+            self.colorbarLabel.hide()
```
If you see my other comment about leaving the adding of the widgets and creating `self.colormapLabel`, then here we just need to add `self.colormapLabel.hide()` and everything should work fine

sofroniewn reviewed (changes requested):
Hi @ekdnam this is coming along very nicely I think. I've requested a couple changes to make us only use the hide functionality.

We'll also want to add a dedicated test for this functionality. Basically you'll want to create one test with a non-rgb layer and check that these widgets are visible, and then you'll want to create another test with an rgb layer and check they are hidden.

You'll want to put the two new tests in a new file called `test_qt_image_layer.py` which will be very similar to this file, but which will for now only contain those two tests. https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/_tests/test_qt_image_base_layer_.py

ekdnam committed:
modify colormapLabel

colormapLabel has now been made an attribute of the base class QtBaseImageControls

ekdnam committed:
in case of RGB image disable colormapComboBox

if the image is of the RGB color model then in such a case, the colormapComboBox is disabled, and only the text "RGB" is displayed

ekdnam committed:
added attribute to class and a condition for rgb image

attribute: colormapLabel, and when there is a rgb image, the combobox displays the text 'rgb'

ekdnam committed:
add hide colorbar condition

in case of an rgb image, the colorbar would be hidden

ekdnam committed:
add parent=self

ekdnam committed:
add QLabel for rgb images

Commit:
Merge branch 'master' of https://github.com/napari/napari into rgb_colormap

Commit:
update rgb colormap display

ekdnam committed:
Merge pull request #2 from napari/master

Rebasing with master

liu-ziyang committed:
Merge branch 'master' into rgb_colormap

Commit:
add line back

Commit:
Merge remote-tracking branch 'ziyang/rgb_colormap' into rgb_colormap

Commit:
merge with branch

ekdnam committed:
Merge pull request #1 from ziyangczi/rgb_colormap

Rgb colormap

liu-ziyang commented on napari/_qt/layer_controls/qt_surface_controls.py:
```diff
@@ -36,7 +36,7 @@ def __init__(self, layer):
         self.grid_layout.addWidget(QLabel('gamma:'), 2, 0)
         self.grid_layout.addWidget(self.gammaSlider, 2, 1)
-        self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
+        self.grid_layout.addWidget(self.colormapLabel, 3, 0)
```
I believe this is the line causing issue, there is no colormapLabel in the object, you may want to revert it

ekdnam committed:
remove object reference

sofroniewn reviewed (approved):
Yes this looks good to me. Here's what the final product looks like, very nice.

Thanks for the work here @ekdnam and thanks @ziyangczi for help finishing this up! I will merge now

<img width="1200" alt="Screen Shot 2020-10-20 at 8 28 52 AM" src="https://user-images.githubusercontent.com/6531703/96608585-656a3200-12ae-11eb-9eb3-5954a6f6075f.png">
